nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoCorsCsrfFilter.java

```diff
@@ -172,6 +230,99 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         response.sendError(HttpServletResponse.SC_FORBIDDEN, message);
     }
 
+    /**
+     * Check safe method according to RFC 7231 4.2.1.
+     */
+    protected boolean isSafeMethod(String method) {
+        return GET.equals(method) || HEAD.equals(method) || OPTIONS.equals(method) || TRACE.equals(method);
```

Comment:
Alternatively you could have a static constant `Set<String>` and do a `contains()`.  It would go nicely when we can use a `Set.of`.